nemo_skills/inference/generate_solutions.py

```diff
@@ -130,15 +131,37 @@ def generate_solutions(cfg: GenerateSolutionsConfig):
         for idx, data_point in tqdm(enumerate(data), initial=starting_idx, total=len(data) + starting_idx):
             if idx >= cfg.max_samples:
                 break
-
             data_points.append(data_point)
 
-            if len(data_points) == cfg.batch_size:
+            if cfg.inference.use_batch_api and len(data_points) == len(data):
+                # Using batch API for generation
+                # Accessing the model directly since CodeExecutionWrapper does not support batch_generate
+                request_metadata = llm.model.batch_generate(
+                    prompts=[prompt.build_string(data_point) for data_point in data_points],
+                    tokens_to_generate=cfg.inference.tokens_to_generate,
+                )
+                with open(cfg.output_file, 'at', encoding='utf-8') as fout:
+                    for data_point in data_points:
+                        fout.write(json.dumps(data_point) + '\n')
+
+                # saving the request id to be able to retrieve results when they are ready
+                with open(cfg.output_file + '-batch-request-id', 'wt', encoding='utf-8') as batch_fout:
+                    batch_fout.write(
+                        json.dumps({'request_id': request_metadata.id, "generation_key": cfg.generation_key})
+                    )
+                LOG.info('Submitted batch evaluation request. Please wait for the results to be ready.')
+                LOG.info('The current status and final results can be accessed through summarize_results.py')
+                LOG.info('Request metadata: %s', str(request_metadata))
+
+                # Clear data_points as we've submitted the batch
+                data_points = []
+
+            elif not cfg.inference.use_batch_api and (len(data_points) == cfg.batch_size or idx == len(data) - 1):
```

Comment:
Let's use cfg.max_samples instead of len(data)

Reply:
Done